### PR TITLE
enable TLS as default configuration (self-signed-certificate)

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: service/http/service.go [hunk 1]
```diff
@@ -105,9 +107,33 @@ func createAuthFunc(config configHttp.Config, clientApplicationServer *grpc.Clie
...
+func SaveConnInContext(ctx context.Context, c net.Conn) context.Context {
+	return context.WithValue(ctx, ConnContextKey, c)
+}
+
+func GetTLSConn(r *http.Request) (*tls.Conn, bool) {
+	c, ok := r.Context().Value(ConnContextKey).(*tls.Conn)
+	return c, ok
+}
+
 // New creates new HTTP service
```
Comment (Danielius1922) [inline service/http/service.go:125]: HTTP/HTTPS

File: pkg/net/listener/tls/listener.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	hdr, err := bconn.buf.Peek(n)
+	_ = c.SetReadDeadline(time.Time{})
+	if err != nil {
+		l.logger.Warnf("closing connection for error while peeking for first 5bytes in 1sec: %v", err)
+		_ = bconn.Close()
+		return
+	}
+
+	// I don't remember what the TLS handshake looks like,
+	// but this works as a POC
```
Comment (Danielius1922) [inline pkg/net/listener/tls/listener.go:74]: is this POC version OK for master, or do we want a better version?

Comment (jkralik) [reply pkg/net/listener/tls/listener.go:74]: Oh this is just copy from Stackoverflow. It is whole reworked :D

File: cmd/generateSelfSigned.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+		return fmt.Errorf("cannot generate key: %w", err)
+	}
+	template := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			Organization: []string{"plgd.dev"},
+			CommonName:   cn,
+		},
+		NotBefore: time.Now(),
+		NotAfter:  time.Now().Add(time.Hour * 24 * 356),
```
Comment (Danielius1922) [inline cmd/generateSelfSigned.go:136]: 356 -> 365

**Review by Danielius1922 — APPROVED** [review]

**Pull request merged**